tasks/tasks/data_modeling.py

```diff
@@ -0,0 +1,398 @@
+        lon_annotation, = filter(lambda a: a['geo_type'] == 'longitude', geo_annotations)
+
+        # load the netcdf and convert coordinate names to time, lat, lon
+        dataset = xr.open_dataset(file_path)
+        dataset = dataset.rename({time_annotation['name']: 'time', lat_annotation['name']: 'lat', lon_annotation['name']: 'lon'})
+
+        # ensure the time dimension is of type cftime.DatetimeNoLeap
+        if isinstance(dataset['time'].values[0], str):
+            # time_annotation['time_format'] # e.g. '%Y-%m-%d %H:%M:%S'
+            times = [pd.to_datetime(t, format=time_annotation['time_format']) for t in dataset['time'].values]
+            times = [DatetimeNoLeap(t.year, t.month, t.day, t.hour, t.minute, t.second) for t in times]
+            dataset['time'] = times
+        elif isinstance(dataset['time'].values[0], DatetimeNoLeap):
+            pass
+        elif isinstance(dataset['time'].values[0], np.datetime64):
+            def to_cftime(t: np.datetime64) -> DatetimeNoLeap:
+                dt = pd.to_datetime(t)
+                return DatetimeNoLeap(dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second)
+            times = [to_cftime(t) for t in dataset['time'].values]
+            dataset['time'] = times
+        else:
+            raise Exception('unhandled time type', type(dataset['time'].values[0]))
+
+
+        # store dataset in map
+        datasets[dataset_id] = dataset
+
+
+    #create loaders for each variable
+    for feature in features:
+        data_source = feature['data_source']
+        variable_name, dataset_id = data_source.split('::')
+        data = datasets[dataset_id][variable_name]
+        time_regrid_type = RegridType[feature['time_aggregation_function']]
+        geo_regrid_type = RegridType[feature['geo_aggregation_function']]
+        loaders[data_source] = lambda: Variable(data, time_regrid_type, geo_regrid_type)
+
+    # clean up downloaded netcdfs
+    shutil.rmtree(tmpdir)
+
+    return loaders
+
+dtype_kind_map = {
+    'b': 'bool',
+    'i': 'int',
+    'u': 'int',
+    'f': 'float',
+}
+
+def post_data_to_dojo(data: xr.Dataset, name: str, dataset_description:str, feature_description:str):
+    print(f'######### Saving to dojo:\n{name}\n{dataset_description}\n{feature_description}\n{data}\n#########')
+    #create indicator
+    metadata = {
+        "name": name,
+        "description": dataset_description,
+        "domains": ["Logic"],
+        "maintainer": {
+            "name": "TODO: who derived the dataset",
+            "email": "todo@jataware2.com",
+            "organization": "Jataware2",
+            "website": "jataware.com"
+        },
+        "file_metadata": {
+            "filetype": "nc"
+        },
+        #TODO: this probably isn't the right way to set the filedata stuff 
+        # it looks like the endpoint is supposed to be setting it, but that wasn't working, while hardcoding it here did
+        "fileData": {
```

Comment:
Yeah- I think this object and properties come from a misunderstanding I had when working on my first project here- porting spaceTag to Dojo. You'll notice fileData, rawFileName use camelcase, since the UI js code is setting that on its "internal" state, which is added to a dataset object, and then sent to the server to be merged (patched) onto the indicator object onto DB (es). I'm not sure if we had all the necessary info before fileData was added. Maybe we did, or maybe it needed to be set by the API when creating indicators.

For now it is out of scope to address, and Ok to do this manually here to create consistency between indicators created using the UI flow and this task.